docs/designs/ps_design.md

```diff
@@ -1,377 +1,147 @@
+* Communication and I/O bandwidth between workers and the PS may limit the training speed. Multiple workers may pull parameters or push gradients at the same time. However, the bandwidth of a single machine is limited. Thus, with single PS instance, communication may be the bottleneck, especially when using a large number of workers or using a big model. This insipres us to use multiple PS instances.
 
-### Compoments
+Therefore, ElasticDL proposes to implement a PS with multiple PS instances and adopt model parallelism by placing each prarameter of the model on one of the PS instances. Such distribution is known as model sharding. In this setup, we can solve the first problem and alleviate the second problem:
 
-![pserver](../images/pserver.png)
+* For models with large embedding tables, every PS instance only contains a subset of embedding parameters. Every worker only contains all non-embedding parameters and embedding parameters that are used in one iteration, which only account for a small proportion of the embedding parameters. Thus, parameters on every PS instance and every worker can fit in the memory of single process.
+* Everytime workers pull parameters (or push gradients), it communicates with all the PS instances to get all model parameters (or push all gradients). Thus if the PS have *N* instances, each worker only uses *1/N* bandwidth of each PS instance. Therefore, the PS with multiple PS instances can alleviate the communication bottleneck.
 
-PS contains two main compoments:
+Also, using the PS with multiple PS instances can benefit performance through accelerating communication, because workers pull parameters concurrently from all the PS nodes, rather than pulling them serially in the single PS node setting.
 
-- KVStore
-- Optimizer
+In the literature, we see works about parameter server designs that handle large models and improve performance. However, ElasticDL needs an additional property -- fault-tolerance. The rest of the document will focus on how to achieve the three goals simultaneously:
 
-The worker initializes a model, and pushes parameters to KVStore. Before each step of training, the worker pulls the latest model from KVStore. After a round of forward/backward computation, the worker pushes gradients to the PS waiting for processing. Then, optimizer of PS will look up the corresponding parameter from KVStore. At last, it applies gradients to parameters, and updates parameter back to KVStore.
+1. model sharding
+2. high performance
+3. fault-tolerance
 
+## Model Sharding
+There are two kinds of parameters in the training process:
 
-### KVStore
+* Embedding parameters consist of mutliple embedding tables. Each embedding table corresponds to one embedding layer in model structure. An embedding table is a data structure that maps a discrete value, named embedding id *id*, to a 1-d vector, named embedding vector *vector*. 
```

Comment:
A vector is 1-D, so no need to say "1-d vector"; just "vector" would be fine. Also, it is 1-D, but not 1-d.

Reply:
Fixed.